test/unit/krane/kubernetes_resource/stateful_set_test.rb

```diff
@@ -4,39 +4,90 @@
 class StatefulSetTest < Krane::TestCase
   include ResourceCacheTestHelper
 
-  def test_deploy_succeeded_is_true_when_revision_and_replica_counts_match
-    template = build_ss_template(status: { "observedGeneration": 2 })
-    ss = build_synced_ss(template: template)
+  def test_deploy_succeeded_when_revision_matches_for_rollingupdate_strategy_with_none_annotation
+    ss_template = build_ss_template(updateStrategy: "RollingUpdate", rollout: "none")
+    ss = build_synced_ss(ss_template: ss_template)
     assert_predicate(ss, :deploy_succeeded?)
   end
 
+  def test_deploy_succeeded_when_revision_matches_for_ondelete_strategy_with_none_annotation
+    ss_template = build_ss_template(updateStrategy: "OnDelete", rollout: "none")
+    ss = build_synced_ss(ss_template: ss_template)
+    assert_predicate(ss, :deploy_succeeded?)
+  end
+
+  def test_deploy_does_not_succeed_when_revision_does_not_match_with_none_annotation
+    ss_template = build_ss_template(status: { "updateRevision": 1 }, rollout: "none")
+    ss = build_synced_ss(ss_template: ss_template)
+    refute_predicate(ss, :deploy_succeeded?)
+  end
+
+  def test_deploy_succeeded_when_replica_counts_match_for_ondelete_strategy_with_full_annotation
+    ss_template = build_ss_template(updateStrategy: "OnDelete", rollout: "full")
+    ss = build_synced_ss(ss_template: ss_template)
+    assert_predicate(ss, :deploy_succeeded?)
+  end
+
+  def test_deploy_does_not_succeed_when_replica_counts_do_not_match_for_ondelete_strategy_with_full_annotation
+    ss_template = build_ss_template(status: { "readyReplicas": 1 }, updateStrategy: "OnDelete", rollout: "full")
+    ss = build_synced_ss(ss_template: ss_template)
+    refute_predicate(ss, :deploy_succeeded?)
+  end
+
+  def test_deploy_succeeded_when_replica_counts_match_for_rollingupdate_strategy_with_full_annotation
+    ss_template = build_ss_template(updateStrategy: "RollingUpdate", rollout: "full")
+    ss = build_synced_ss(ss_template: ss_template)
+    assert_predicate(ss, :deploy_succeeded?)
+  end
+
+  def test_deploy_does_not_succeed_when_replica_counts_do_not_match_for_rollingupdate_strategy_with_full_annotation
+    ss_template = build_ss_template(status: { "currentReplicas": 1 }, updateStrategy: "RollingUpdate", rollout: "full")
+    ss = build_synced_ss(ss_template: ss_template)
+    refute_predicate(ss, :deploy_succeeded?)
+  end
+
   def test_deploy_failed_ensures_controller_has_observed_deploy
```

Comment:
Not sure whats being tested here

Reply:
good catch - this was an existing test to confirm that `deploy_succeeded?` is **not** true when current and observed generations aren't equal. Will update the test name to be clearer